src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
I'm ok w/ this approach.  The reality is that HttpMethod will only have ascii characters in them since it is designed to be used for HTTP.